channeldb/invoices.go

```diff
@@ -1148,38 +1145,42 @@ func fetchInvoice(invoiceNum []byte, invoices kvdb.RBucket,
 		return invpkg.Invoice{}, err
 	}
 
-	// If this is an AMP invoice, then we'll also attempt to read out the
-	// set of HTLCs that were paid to prior set IDs. However, we'll only do
-	// this is the invoice didn't already have HTLCs stored in-line.
-	invoiceIsAMP := invoice.Terms.Features.HasFeature(
-		lnwire.AMPOptional,
-	)
-	switch {
-	case !invoiceIsAMP:
-		return invoice, nil
+	// If this is an AMP invoice we'll also attempt to read out the set of
+	// HTLCs that were paid to prior set IDs, if needed.
+	if invoice.IsAMP() && shouldFetchAMPHTLCs(invoice, setIDs) {
+		invoice.Htlcs, err = fetchAmpSubInvoices(
+			invoices, invoiceNum, setIDs...,
+		)
+		// TODO(positiveblue): should we fail when we are not able to
```

Comment:
I think we probably should. Since we know it's an AMP invoice here, we do want its data to be loaded correctly.

Reply:
I agree. I wasn’t sure about adding it in this commit it not because it is a breaking change. It was weird for me to see that we are currently ignoring this error.

Reply:
Yep, the failure also matches the prior behavior.

Reply:
I think it was a bug and we did intend to return the error but we were just ignoring it?

https://github.com/lightningnetwork/lnd/blob/master/channeldb/invoices.go#L1178

That's why I added the log + the comment. If it is a bug this PR is a good place to fix it

Reply:
Sounds good to me. Just a final nit: add newline before comment?